central/auth/datastore/datastore_impl_test.go

```diff
@@ -0,0 +1,240 @@
+//go:build sql_integration
+
+package datastore
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	pgStore "github.com/stackrox/rox/central/auth/datastore/internal/store/postgres"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stackrox/rox/pkg/postgres/pgtest"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/sac/resources"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestAuthDatastorePostgres(t *testing.T) {
+	suite.Run(t, new(datastorePostgresTestSuite))
+}
+
+type datastorePostgresTestSuite struct {
+	suite.Suite
+
+	ctx       context.Context
+	pool      *pgtest.TestPostgres
+	store     pgStore.Store
+	datastore DataStore
+}
+
+func (s *datastorePostgresTestSuite) SetupTest() {
+	s.ctx = sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS, storage.Access_READ_WRITE_ACCESS),
+			sac.ResourceScopeKeys(resources.Access),
+		),
+	)
```

Comment:
Yeah, they are a part of [the generic store now](https://github.com/stackrox/stackrox/blob/master/pkg/search/postgres/store.go#L445-L453). Previously this was required to be handled within each datastore.

Reply:
Unresolved: something fishy is happening here.
I tried commenting out these lines and the test still passes.
```go
			// sac.AccessModeScopeKeys(storage.Access_READ_ACCESS, storage.Access_READ_WRITE_ACCESS),
			// sac.ResourceScopeKeys(resources.Access),
```

Is it the same for you? Do you know why is that?

Reply:
Does for me too.

If you switch to a different resource, you will get an resource access denied error.

The reason is the semantics of `sac.AllowFixedScopes`. In case the list is empty, _all_ access will be assumed, hence you won't get any errors.

Ref: https://github.com/stackrox/stackrox/blob/master/pkg/sac/allow_fixed_scopes_checker_core.go#L163

Reply:
I know nothing about our authz implementation, but that's very unintuitive.

However, this can be resolved now, thanks for educating me.